File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(),points.end(), [](const vector<int> &o1, const vector<int> &o2) {
            return (o1[1] < o2[1]);
        });
        int arrows=1,xStart,xEnd,firstEnd=points[0][1];
        for(auto point:points){
            xStart=point[0];
            xEnd=point[1];
            if (firstEnd<xStart){
                arrows++;
                firstEnd=xEnd;
            }
        }
        return arrows;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x:x[1])
        first_end,arrows=points[0][1],1
        for point_start,point_end in points:
            if first_end<point_start:
                arrows+=1
                first_end=point_end
        return arrows